File: src/types/printer.ts
export interface Printer {
  id:           string;
  name:         string;
  printer_type: "thermal_tcp" | "screen";
  ip:           string | null;
  port:         number;
  paper_mm:     number;
  roles:        string;  // "receipt" | "kitchen" | "receipt,kitchen"
  sort_order:   number;
  created_at:   string;
}

export interface PrinterInput {
  name:         string;
  printer_type: "thermal_tcp" | "screen";
  ip:           string | null;
  port:         number;
  paper_mm:     number;
  roles:        string;
}

export function printerHasRole(p: Printer, role: "receipt" | "kitchen"): boolean {
  return p.roles.split(",").map((r) => r.trim()).includes(role);
}

File: src-tauri/src/db/models/printer.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Serialize, FromRow)]
pub struct Printer {
    pub id:           String,
    pub name:         String,
    pub printer_type: String,
    pub ip:           Option<String>,
    pub port:         i64,
    pub paper_mm:     i64,
    pub roles:        String,
    pub sort_order:   i64,
    pub created_at:   String,
}

#[derive(Debug, Deserialize)]
pub struct PrinterInput {
    pub name:         String,
    pub printer_type: String,
    pub ip:           Option<String>,
    pub port:         i64,
    pub paper_mm:     i64,
    pub roles:        String,
}
